compiler/sigmatch.nim

```diff
@@ -1091,7 +1091,7 @@ proc typeRel(c: var TCandidate, f, aOrig: PType,
 
   template considerPreviousT(body: untyped) =
     var prev = PType(idTableGet(c.bindings, f))
-    if prev == nil: body
+    if prev == nil or prev == f: body
```

Comment:
Your PR is great! Unfortunately I don't understand it. How could it be that ``prev == f``?

Reply:
Good point.  I had only fixed the symptom.  `TilesetCT` was being compared against itself and `trDontBind` wasn't being preserved.  This is now fixed.